app/src/main/java/com/github/ybecker/epforuml/authentication/MockAuthenticator.kt

```diff
@@ -7,18 +7,34 @@ import com.github.ybecker.epforuml.database.DatabaseManager
 import com.github.ybecker.epforuml.database.Model
 
 class MockAuthenticator(private val activity: AppCompatActivity) : Authenticator {
+    private val userId = "0"
+    private val userName = "TestUser"
+    private val email = "testEmail"
+
     override fun signIn() {
-        val futureUser = DatabaseManager.getDatabase().addUser("0", "TestUser", "testEmail")
-        futureUser.thenAccept { DatabaseManager.user = futureUser.get() }
-        activity.startActivity(Intent(activity, MainActivity::class.java))
+        val futureUser =
+            DatabaseManager.getDatabase().addUser(userId, userName, email)
+        futureUser.thenAccept {
+            DatabaseManager.user = it
+            DatabaseManager.db.setUserPresence(it.userId)
+            DatabaseManager.user!!.connections.add(true)
+            activity.startActivity(Intent(activity, MainActivity::class.java))
+        }
     }
 
```

Comment:
For signout and delete you could do a  private function than could be called in these to function instead of repeating the code